### ISSUE TITLE
A way to remove all invalid recent repositories from the recent repositories list

### ISSUE THREAD

**Issue by lhiginbotham** [gitextensions/gitextensions#5616]
**Do you want to request a *feature* or report a *bug*?**
feature

**What is the current behavior?**
When clicking on an entry in the recent repositories list, if the entry is not a valid git repository (such as if the repo was moved from the location, or the location no longer exists) then a window pops up asking if you want the entry removed from the recent repositories list.

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**

**What is the expected behavior?**
A new button/option to remove all invalid git repositories from the recent repositories list.  This could be in the same popup window, and another button could exist in the recent repositories list's configuration page.

**Environment you encounter the issue:**
Windows

- GitExtensions version: 3.0
- GIT version: 
- OS version:  Windows 10
- .NET version: 

**Did this work in previous version of GitExtensions (which)?**

**Comment by lhiginbotham** [comment]
I am interested in helping out with GitExtensions.  If this feature request is approved, would it be a good first issue for me to get started with?

**Comment by lhiginbotham** [comment]
@RussKie 
Is my request similar to your request?:

Dashboard: bulk delete entries #5512
(https://github.com/gitextensions/gitextensions/issues/5512)

**Comment by lhiginbotham** [comment]
@drewnoakes, I had something like that in mind for the dialog that pops up when you try to select an invalid repo from the repo list, so I would like to move forward with your mockup.

As for the recent repositories settings dialog, I would like to know whether or not a button like this would be appropriate:

![image](https://user-images.githubusercontent.com/14078392/47451919-f4631300-d796-11e8-8880-031326e50494.png)

On one hand, it would be nice to have a button always be visible that informs the user whether or not any of their recent repositories are no longer valid.  On the other hand, it may be seen as unnecessary clutter in the UI, and we would rather prefer to only present the option to remove all invalid repos to the user when they are trying to access an invalid repo.

**Comment by lhiginbotham** [comment]
@drewnoakes I couldn't go ahead with your proposal as the "directory is not a valid repository" dialog box is a primitive System.Windows.Forms.MessageBox.  As such, I could not create a new button for the dialog box, and am constrained to using the dialog button sets defined in the System.Windows.Forms.MessageBoxButtons enum.  I decided to introduce a second dialog box that pops up when the user selects "yes - remove the invalid repo" on the first dialog box.  This second dialog box alerts the user that there are multiple invalid repositories, and offers to remove all of them.  The user is presented with a Yes or a No option.  Selecting Yes removes all invalid repos, while selecting No goes ahead with removing the originally clicked repo.

First dialog box when clicking an invalid repo (unchanged, currently in the product):

![first_dialog](https://user-images.githubusercontent.com/14078392/47474923-58152c80-d7e7-11e8-954b-34fcf5231fd3.JPG)

New dialog box when there are multiple invalid repos in your list:

![image](https://user-images.githubusercontent.com/14078392/47474951-74b16480-d7e7-11e8-9f1b-185aaf82f107.png)

**Comment by RussKie** [comment]
> Is my request similar to or a subset of your previous request?:

I would think it is not quite a dup but very similar, #5512 is more along the line of "select and delete".

I like the @drewnoakes's proposal. If we can't achieve this with the `MessageBox`,  I'm sure we can achieve this with `PSTaskDialog.cTaskDialog` - please have a look through the codebase to see how it is used.

WRT: "Recent repositories" dialog - I'd like to remove it altogether and move required configs to the main configuration dialog. It is on todo list.

> always be visible that informs the user whether or not any of their recent repositories are no longer valid. 

This information is available on the dashboard, an invalid repo is marked with a ⚠️ icon. Unfortunately this information is not available outside the dashboard. Perhaps we need to look at bridging this gap.

**Comment by vbjay** [comment]
There "are"

**Comment by drewnoakes** [comment]
I think "is" is correct, because there's one total. However I agree it's hard to parse this sentence.

How about "There are {0} invalid repositories in thre recent repositories list.\n\nDo you want to remove all of them?"

We only show this when N > 1 so the plural form should suffice alone.

**Comment by vbjay** [comment]
Use a turnary

$"There {count>1 ? "are":"is"} ...

On phone might need to add ().

**Comment by drewnoakes** [comment]
> We only show this when N > 1 so the plural form should suffice alone.

**Comment by vbjay** [comment]
Durr. Lol. Haven't had my coffee.

**Comment by vbjay** [comment]
Actually. If there are 2 to start off with they hit yes and knock it down
to 1 then the singular would be valid.

**Comment by RussKie** [comment]
LOL, use `Strings` approach
https://github.com/gitextensions/gitextensions/blob/8f739f0054e5b87da70d5753251f3de00242803d/ResourceManager/Strings.cs#L45-L50
